FormalConjectures/ErdosProblems/107.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 107
+
+*Reference:* [erdosproblems.com/107](https://www.erdosproblems.com/107)
+-/
+
+open Filter
+
+namespace Erdos107
+
+abbrev Point := EuclideanSpace ℝ (Fin 2)
+
+namespace Point
+
+/-- The three points `a,b,c` are not collinear. -/
+def InGenPos₃ (a b c : Point) : Prop :=
+  Matrix.det !![a 0, b 0, c 0 ; a 1, b 1, c 1 ; 1, 1, 1] ≠ 0
+
+/-- No three points in `S` are collinear. -/
+def SetInGenPos (S : Set Point) : Prop :=
+  ∀ ⦃p q r : Point⦄, {p,q,r} ⊆ S → List.Nodup [p,q,r] → InGenPos₃ p q r
+
+end Point
+
+/-- The set `S` is in convex position. -/
+def ConvexIndep (S : Set Point) : Prop :=
+  ∀ a ∈ S, a ∉ convexHull ℝ (S \ {a})
+
+/-- The set `P` contains a convex `n`-gon. -/
+def HasConvexNGon (n : ℕ) (P : Set Point) : Prop :=
+  ∃ S : Finset Point, S.card = n ∧ ↑S ⊆ P ∧ ConvexIndep S
+
+noncomputable def f (n : ℕ) : ℕ :=
+  sInf { N | ∀ (pts : Finset Point), pts.card = N → Point.SetInGenPos pts →
+    HasConvexNGon n pts }
+
+/--
+Let $f(n)$ be minimal such that any $f(n)$ points in $ℝ^2$, no three on a line,
+contain $n$ points which form the vertices of a convex $n$-gon.
+Prove that $f(n) = 2^{n-2} + 1$.
+-/
+@[category research open, AMS 52]
+theorem erdos_107 : (∀ n ≥ 3, f n = 2^(n-2) + 1) ↔ answer(sorry) := by
+  sorry
+
+/-- Depending on details of definitions,
+the statement is false or trivial for $n < 3$. -/
+@[category test, AMS 52]
+theorem erdos_107.f_zero_eq : f 0 = 0 := by
+  have : ∀ P, HasConvexNGon 0 P := by
+    intro; use ∅; simp [ConvexIndep]
+  simp [f, this]
```

Comment:
Another good test statement here might be the case $n=3$.

Reply:
I added the statement. I also started filling in a proof, but it's getting a bit long; I understand you don't want those in this repo. (We have a proof modulo assuming that some CNF is unsatisfiable [here](https://github.com/bsubercaseaux/EmptyHexagonLean/blob/main/Lean/Geo.lean#L46).)